command_line_assistant/utils/files.py

```diff
@@ -93,3 +100,105 @@ def write_file(contents: Union[str, bytes], path: Path, mode: int = 0o600) -> No
             path,
             str(e),
         )
+
+
+class NamedFileLock:
+    """File-based lock mechanism for general locking purposes.
+
+    This class provides a way to track state across processes
+    using a lock file. The lock file contains the PID of the capturing process.
+
+    Usage:
+        with NamedFileLock(name="my_file") as lock:
+            # Execute your operations
+        # Lock is automatically released
+    """
+
+    def __init__(self, name: str) -> None:
+        """Initialize the named file lock mechanism."""
+        self._pid: int = os.getpid()
+        self._name: str = name.replace("-", "_")
+        self._lock_file: Path = Path(get_xdg_state_path(), name + ".lock")
+
+    @property
+    def pid(self) -> int:
+        """Return the parent pid."""
+        return self._pid
+
+    @pid.setter
+    def pid(self, value: int) -> None:
+        """Set the value for the internal _pid property."""
+        self._pid = value
```

Comment:
This property is new, but I can't see where it's used. Did I miss its usage elsewhere?

Reply:
OOH of course. I read right over that when reading the error messages!